src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/resource_allocation_manager_impl.cc

```diff
@@ -0,0 +1,506 @@
+#include "rc_rpc_plugin/resource_allocation_manager_impl.h"
+#include "application_manager/application.h"
+#include "application_manager/application_manager.h"
+#include "interfaces/HMI_API.h"
+#include "interfaces/MOBILE_API.h"
+#include "smart_objects/enum_schema_item.h"
+#include "rc_rpc_plugin/rc_rpc_plugin.h"
+#include "application_manager/message_helper.h"
+#include "rc_rpc_plugin/rc_module_constants.h"
+#include "json/json.h"
+#include "utils/helpers.h"
+#include "utils/make_shared.h"
+
+//#include "remote_control/message_helper.h"
+//#include "remote_control/rc_app_extension.h"
+
+namespace rc_rpc_plugin {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "RemoteControlModule")
+
+ResourceAllocationManagerImpl::ResourceAllocationManagerImpl(
+    application_manager::ApplicationManager& app_mngr,
+    application_manager::rpc_service::RPCService& rpc_service)
+    : current_access_mode_(hmi_apis::Common_RCAccessMode::AUTO_ALLOW)
+    , app_mngr_(app_mngr)
+    , rpc_service_(rpc_service)
+    , is_rc_enabled_(true) {}
+
+ResourceAllocationManagerImpl::~ResourceAllocationManagerImpl() {}
+
+AcquireResult::eType ResourceAllocationManagerImpl::AcquireResource(
+    const std::string& module_type, const uint32_t app_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const application_manager::ApplicationSharedPtr acquiring_app =
+      app_mngr_.application(app_id);
+  if (!acquiring_app) {
+    LOG4CXX_WARN(logger_, "App with app_id: " << app_id << "does not exist!");
+    return AcquireResult::IN_USE;
```

Comment:
why return AcquireResult::IN_USE when we cannot find the app in the application_manager?